teuthology/lock.py

```diff
@@ -226,6 +226,13 @@ def main(ctx):
     if ctx.num_to_lock:
         assert ctx.machine_type, \
             'must specify machine type to lock'
+    if ctx.machine_type == 'vps':
+        if ctx.os_type is None:
+            ctx.os_type=random.choice(default_table.keys())
+	    ctx.os_version=random.choice(default_table[ctx.os_type])
```

Comment:
@zmc: I think the function does more than just return table of valid distros, if you use the teuthology-lock cli it creates VMs. After I pushed this PR i realized this is not the best place since it handles only  "teuthology-lock" cli case, probably Downburst class may be the right place if no other function is setting default values before they call the init or init can set the default values to be more clear and should also own supported_table dict. I will close this and reopen a different one a